pkg/azurefile/controllerserver.go

```diff
@@ -825,68 +826,130 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		mc.ObserveOperationWithResult(isOperationSucceeded, SourceResourceID, sourceVolumeID, SnapshotName, snapshotName)
 	}()
 
-	exists, item, err := d.snapshotExists(ctx, sourceVolumeID, snapshotName, req.GetSecrets())
-	if err != nil {
+	if len(req.GetSecrets()) > 0 {
+		exists, item, err := d.snapshotExists(ctx, sourceVolumeID, snapshotName, req.GetSecrets())
+		if err != nil {
+			if exists {
+				return nil, status.Errorf(codes.AlreadyExists, "%v", err)
+			}
+			return nil, status.Errorf(codes.Internal, "failed to check if snapshot(%v) exists: %v", snapshotName, err)
+		}
+		if exists {
+			klog.V(2).Infof("snapshot(%s) already exists", snapshotName)
+			tp := timestamppb.New(item.Properties.LastModified)
+			if tp == nil {
+				return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", item.Properties.LastModified)
+			}
+			if item.Snapshot == nil {
+				return nil, status.Errorf(codes.Internal, "Snapshot property of %s is nil", item.Name)
+			}
+			return &csi.CreateSnapshotResponse{
+				Snapshot: &csi.Snapshot{
+					SizeBytes:      volumehelper.GiBToBytes(int64(item.Properties.Quota)),
+					SnapshotId:     sourceVolumeID + "#" + *item.Snapshot,
+					SourceVolumeId: sourceVolumeID,
+					CreationTime:   tp,
+					// Since the snapshot of azurefile has no field of ReadyToUse, here ReadyToUse is always set to true.
+					ReadyToUse: true,
+				},
+			}, nil
+		}
+	} else {
+		exists, item, err := d.fileShareSnapshotExists(ctx, sourceVolumeID, snapshotName)
+		if err != nil {
+			if exists {
+				return nil, status.Errorf(codes.AlreadyExists, "%v", err)
+			}
+			return nil, status.Errorf(codes.Internal, "failed to check if snapshot(%v) exists: %v", snapshotName, err)
+		}
 		if exists {
-			return nil, status.Errorf(codes.AlreadyExists, "%v", err)
+			klog.V(2).Infof("snapshot(%s) already exists", snapshotName)
+			tp := timestamppb.New(item.SnapshotTime.Time)
+			if tp == nil {
+				return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", item.SnapshotTime.Time)
+			}
+			if item.SnapshotTime.Format(snapshotTimeFormat) == "" {
+				return nil, status.Errorf(codes.Internal, "Snapshot property of %s is nil", *item.Name)
+			}
+			return &csi.CreateSnapshotResponse{
+				Snapshot: &csi.Snapshot{
+					SizeBytes:      volumehelper.GiBToBytes(int64(*item.ShareQuota)),
+					SnapshotId:     sourceVolumeID + "#" + item.SnapshotTime.Format(snapshotTimeFormat),
```

Comment:
use a var to assign `item.SnapshotTime.Format(snapshotTimeFormat)` since L871 already does the transform.